x/txfees/keeper/feedecorator_test.go

```diff
@@ -226,11 +189,13 @@ func (suite *KeeperTestSuite) TestFeeDecorator() {
 			_, err := antehandlerMFD(suite.Ctx, tx, false)
 
 			if tc.expectPass {
-				if tc.baseDenomGas && !tc.txFee.IsZero() {
-					moduleAddr := suite.App.AccountKeeper.GetModuleAddress(types.FeeCollectorName)
-					suite.Require().Equal(tc.txFee[0], suite.App.BankKeeper.GetBalance(suite.Ctx, moduleAddr, baseDenom), tc.name)
-				} else if !tc.txFee.IsZero() {
-					moduleAddr := suite.App.AccountKeeper.GetModuleAddress(types.NonNativeFeeCollectorName)
+				// ensure fee was collected
+				if !tc.txFee.IsZero() {
+					moduleName := types.FeeCollectorName
+					if tc.txFee[0].Denom != baseDenom {
+						moduleName = types.NonNativeFeeCollectorName
+					}
```

Comment:
Base denom gas was an unnecessary field, its role is automated with this if statement.